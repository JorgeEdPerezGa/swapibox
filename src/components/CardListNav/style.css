.card-list-nav-button {
  height: 30px;
  width: 30px;
  border: none;
  margin: 0 12px 0 12px; }
  .card-list-nav-button:hover {
    cursor: pointer; }
  .card-list-nav-button:disabled {
    cursor: auto; }
  .card-list-nav-button:focus {
    outline: none; }

.first-button {
  background-image: url("../../assets/first.svg"); }
  .first-button:hover {
    background-image: url("../../assets/first-hover.svg"); }
  .first-button:disabled {
    background-image: url("../../assets/first-disabled.svg"); }

.previous-button {
  background-image: url("../../assets/previous.svg"); }
  .previous-button:hover {
    background-image: url("../../assets/previous-hover.svg"); }
  .previous-button:disabled {
    background-image: url("../../assets/previous-disabled.svg"); }

.next-button {
  background-image: url("../../assets/next.svg"); }
  .next-button:hover {
    background-image: url("../../assets/next-hover.svg"); }
  .next-button:disabled {
    background-image: url("../../assets/next-disabled.svg"); }

.last-button {
  background-image: url("../../assets/last.svg"); }
  .last-button:hover {
    background-image: url("../../assets/last-hover.svg"); }
  .last-button:disabled {
    background-image: url("../../assets/last-disabled.svg"); }
