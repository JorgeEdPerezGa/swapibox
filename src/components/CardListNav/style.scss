.card-list-nav-button {
  height: 30px;
  width: 30px;
  border: none;
  margin: 0 12px 0 12px;
  background: none;
  &:hover {
    cursor: pointer;
  }
  &:disabled {
    cursor: auto;
  }
  &:focus {
    outline: none;
  }
}

.first-button {
  background-image: url('../../assets/first.svg');
  &:hover {
    background-image: url('../../assets/first-hover.svg');
  }
  &:disabled {
    background-image: url('../../assets/first-disabled.svg');
  }
}

.previous-button {
  background-image: url('../../assets/previous.svg');
  &:hover {
    background-image: url('../../assets/previous-hover.svg');
  }
  &:disabled {
    background-image: url('../../assets/previous-disabled.svg');
  }
}

.next-button {
  background-image: url('../../assets/next.svg');
  &:hover {
    background-image: url('../../assets/next-hover.svg');
  }
  &:disabled {
    background-image: url('../../assets/next-disabled.svg');
  }
}

.last-button {
  background-image: url('../../assets/last.svg');
  &:hover {
    background-image: url('../../assets/last-hover.svg');
  }
  &:disabled {
    background-image: url('../../assets/last-disabled.svg');
  }
}
