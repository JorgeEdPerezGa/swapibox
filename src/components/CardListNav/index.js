import React from 'react';
import style from './style.css';
import PropTypes from 'prop-types';

const CardListNav = ({ type, fetchFirst, handleFetch, fetchLast }) => {
  const fetch = type;
  const disabledPrevious = !fetch || !fetch.previous;
  const disabledNext = !fetch || !fetch.next;

  return (
    <div>
      <button
        className = 'card-list-nav-button first-button'
        onClick = { () => fetchFirst() }
        disabled = { disabledPrevious }></button>
      <button
        className = 'card-list-nav-button previous-button'
        onClick = { () => handleFetch(fetch.previous) }
        disabled = { disabledPrevious }></button>
      <button
        className = 'card-list-nav-button next-button'
        onClick = { () => handleFetch(fetch.next) }
        disabled = { disabledNext }></button>
      <button
        className = 'card-list-nav-button last-button'
        onClick = { () => fetchLast() }
        disabled = { disabledNext }></button>
    </div>
  );
};

CardListNav.propTypes = {
  type: PropTypes.object,
  fetchFirst: PropTypes.func,
  handleFetch: PropTypes.func,
  fetchLast: PropTypes.func
};

export default CardListNav;
