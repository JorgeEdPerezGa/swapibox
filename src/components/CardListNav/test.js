import React from 'react';
import CardListNav from './index';
import setupTest from '../../setupTests';
import { shallow } from 'enzyme';

describe('<CardListNav/>', () => {
  const wrapper = <CardListNav />;

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
