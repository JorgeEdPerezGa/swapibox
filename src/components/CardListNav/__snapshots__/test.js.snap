// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardListNav/> should match snapshot 1`] = `<CardListNav />`;
